[Discover][docs] Add search for relevance (#90611)
elastic/kibana: docs/discover/search-for-relevance.asciidoc

[[discover-search-for-relevance]]
=== Search for relevance

Sometimes you might be unsure which documents best match your search.
{es} assigns a relevancy, or score to each document, so you can
can narrow you search to the documents with the most relevant results.
The higher the score, the better it matches your query.

For example, suppose you have the <<gs-get-data-into-kibana, sample flights data set>>, and you're a searching for
a flight that arrived or departed from `Warsaw` or `Venice` when the weather was clear.

> * In *Discover*, open the index pattern dropdown, and select `kibana_sample_data_flight`.

Reviewer: ```suggestion
.  In *Discover*, open the index pattern dropdown, and select `kibana_sample_data_flight`.
```

> * In the query bar, click  *KQL*, and switch to the <<lucene-query, Lucene query syntax>>.

Reviewer: ```suggestion
.  In the query bar, click  *KQL*, and switch to the <<lucene-query, Lucene query syntax>>.
```

> * Search for `Warsaw OR Venice OR Clear`.

Reviewer: ```suggestion
. Search for `Warsaw OR Venice OR Clear`.
```

> * If you don't see any results, open the time filter and select a time range that contains data.

Reviewer: ```suggestion
.  If you don't see any results, open the time filter and select a time range that contains data.
```

> * From the list of *Available fields*, add `_score` to the document table.

Reviewer: ```suggestion
.  From the list of *Available fields*, add `_score` to the document table.
```

> * Sort the table by descending scores.

Reviewer: ```suggestion
. Click the header for the `_score` column and sort it  by descending scores.
```

> * Now the result is sorted by `Time` (1) and `_score` (2).  To sort only by `_score` you've

Reviewer: ```suggestion
+
The results are currently sorted by first `Time`, and then by `_score`.
```

>  to remove sorting by `Time`.

Reviewer: ```suggestion
. To sort only by `_score`, remove the `Time` field.
```

> * As a result you've got a table of documents starting with the best matches.

Reviewer: ```suggestion
+
Your table now shows documents with the best matches, from most to least relevant.
```

> [role="screenshot"]

Reviewer: ```suggestion
+
[role="screenshot"]
```

[[discover-search-for-relevance]]

> === Search for relevance

Reviewer 2: If you don't want this page to be nested under the "Search data" page, it needs to be bumped up a level:
```suggestion
== Search for relevance
```

Sometimes you might be unsure which documents best match your question. 
{es} assigns a relevancy, or score to each document, so you can

> can narrow you search to the documents with the most relevant results.

Reviewer: ```suggestion
can narrow your search to the documents with the most relevant results.
```

The higher the score, the better it matches your query.

For example, suppose you have the <<gs-get-data-into-kibana, sample flights data set>>, and you're a searching for
a flight that arrived or departed from `Warsaw` or `Venice` when the weather was clear.

.  In *Discover*, open the index pattern dropdown, and select `kibana_sample_data_flight`.
.  In the query bar, click  *KQL*, and switch to the <<lucene-query, Lucene query syntax>>.
. Search for `Warsaw OR Venice OR Clear`.
.  If you don't see any results, open the time filter and select a time range that contains data.
.  From the list of *Available fields*, add `_score` to the document table.

> . Click the header for the `_score` column and sort it  by descending scores.

Reviewer: ```suggestion
. In the document table, click the header for the `_score` column, and then sort the column by descending scores.
```
